components/results_table.go

```diff
@@ -1375,3 +1372,36 @@ func (table *ResultsTable) UpdateSidebar() {
 
 	}
 }
+
+func (table *ResultsTable) isAnInsertedRow(rowIndex int) (isAnInsertedRow bool, DBChangeIndex int) {
+	for i, dmlChange := range *table.state.listOfDBChanges {
+		values := dmlChange.Values
+
+		for _, value := range values {
+			if value.TableRowIndex == rowIndex {
+				cellReference := table.GetCell(rowIndex, 0).GetReference()
+
+				isAnInsertedRow := cellReference != nil && cellReference.(string) != "NULL&" && cellReference.(string) != "EMPTY&" && cellReference.(string) != "DEFAULT&"
+
+				if isAnInsertedRow {
+					return true, i
+				}
+			}
+		}
+
+	}
+	return false, -1
```

Comment:
I would like to suggest


```suggestion
func (table *ResultsTable) isAnInsertedRow(rowIndex int) (isAnInsertedRow bool, DBChangeIndex int) {
	for i, dmlChange := range *table.state.listOfDBChanges {

		for _, value := range dmlChange.Values {
			if value.TableRowIndex != rowIndex {
				continue
			}

			cellReference := table.GetCell(rowIndex, 0).GetReference()
			if cellReference == nil {
				continue
			}

			switch cellReference.(string) {
				 case "NULL&", "EMPTY&", "DEFAULT&":
					continue
				default:
					return true, i
			}
		}
	}
	return false, -1
```

this is pseudo code written on my phone, please check it

Reply:
But maybe this would be more efficient

```suggestion
func (table *ResultsTable) isAnInsertedRow(rowIndex int) (isAnInsertedRow bool, DBChangeIndex int) {
	for i, dmlChange := range *table.state.listOfDBChanges {

		for _, value := range dmlChange.Values {
			if value.TableRowIndex != rowIndex {
				continue
			}

			cellReference := table.GetCell(rowIndex, 0).GetReference()
			if cellReference == nil {
				break
			}

			switch cellReference.(string) {
				 case "NULL&", "EMPTY&", "DEFAULT&":
				default:
					return true, i
			}
			break
		}
	}
	return false, -1
```